### PR TITLE
[CAS] Fixes for Swift Caching Issues

### PR DESCRIPTION
**Pull request by cachemeifyoucan**
Fix various corner cases is swift caching.

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

File: lib/ClangImporter/ClangImporter.cpp [hunk 1]
```diff
@@ -939,6 +939,22 @@ std::string ClangImporter::getOriginalSourceFile(StringRef PCHFilename) {
...
+void ClangImporter::addClangInvovcationDependencies(
+    std::vector<std::string> &files) {
+  auto addFiles = [&files](const std::vector<std::string> &F) {
+    files.insert(files.end(), F.begin(), F.end());
+  };
+  auto &invocation = *Impl.Invocation;
+  // FIXME: Add file dependencies that are not accounted. The long term solution
+  // is to do a dependency scanning for clang importer and use that directly.
+  addFiles(invocation.getHeaderSearchOpts().VFSOverlayFiles);
+  for (auto &Entry : invocation.getHeaderSearchOpts().UserEntries)
```
Comment (benlangmuir) [inline lib/ClangImporter/ClangImporter.cpp:951]: Why are user entries added here but not system? Is this for headermap files?  Can we trim out directories here?

Comment (cachemeifyoucan) [reply lib/ClangImporter/ClangImporter.cpp:951]: Yeah, this is for header maps. How do I check if the entry is header maps or not? Just `stat` and see if it is a file or a directory?

Comment (benlangmuir) [reply lib/ClangImporter/ClangImporter.cpp:951]: Do you have the full clang compiler instance around? If so you could check SearchDirs or HeaderMaps where it is already resolved.  If you only have options, yeah all you can do is check if it's a directory with stat.

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

**Review by artemcm — APPROVED** [review]

**Review by benlangmuir — APPROVED** [review]

**Pull request merged**